runtime/browser/ssl_error_page.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_SSL_ERROR_PAGE_H_
+#define XWALK_RUNTIME_BROWSER_SSL_ERROR_PAGE_H_
+
+#include <string>
+
+#include "base/callback.h"
+#include "content/public/browser/interstitial_page_delegate.h"
+#include "net/ssl/ssl_info.h"
+#include "url/gurl.h"
+
+namespace content {
+class InterstitialPage;
+class WebContents;
+}
+
+namespace xwalk {
+
+class SSLErrorPage : public content::InterstitialPageDelegate {
+ public:
+  enum UserCommandsFromPage {
+    CMD_DONT_PROCEED = 0,
+    CMD_PROCEED = 1,
+  };
+
+  SSLErrorPage(content::WebContents* web_contents,
+               int cert_error,
+               const net::SSLInfo& ssl_info,
+               const GURL& request_url,
+               const base::Callback<void(bool)>& callback);
+
+  ~SSLErrorPage() override;
+
+  // Show an interstitial page to let user to choose the next action
+  void Show();
```

Comment:
If so, we have to always be careful to put 'interstitial_page_'  as the last member of class 'SSLErrorPage'. Since 'SSLErrorPage' is the delegate of 'interstitial_page_' (its class is 'content::InterstitialPage'), many memebers (e.g. ssl_info_) of  'SSLErrorPage' will be referred (e.g., through GetHTMLContents), so we have make sure all members being initialized before 'interstitial_page_'.